typescript/infra/scripts/gas-oracle/update-storage-gas-oracle.ts

```diff
@@ -0,0 +1,129 @@
+import { ethers } from 'ethers';
+
+import {
+  ChainName,
+  ChainNameToDomainId,
+  DomainIdToChainName,
+  HyperlaneCore,
+} from '@hyperlane-xyz/sdk';
+
+import { RemoteGasData, StorageGasOracleConfig } from '../../src/config';
+import { deployEnvToSdkEnv } from '../../src/config/environment';
+import { RemoteGasDataConfig } from '../../src/config/gas-oracle';
+import { getArgs, getCoreEnvironmentConfig, getEnvironment } from '../utils';
+
+/**
+ * Updates the currently stored gas data on the StorageGasOracle contract
+ * if the configured data differs from the on-chain data.
+ * Expects the deployer key to be the owner of the StorageGasOracle contract.
+ */
+async function main() {
+  const args = await getArgs()
+    .boolean('dry-run')
+    .describe('dry-run', 'If true, will not submit any transactions')
+    .default('dry-run', false).argv;
+
+  const environment = await getEnvironment();
+  const coreEnvConfig = getCoreEnvironmentConfig(environment);
+  const multiProvider = await coreEnvConfig.getMultiProvider();
+
+  const storageGasOracleConfig = coreEnvConfig.storageGasOracleConfig;
+  if (!storageGasOracleConfig) {
+    throw Error(`No storage gas oracle config for environment ${environment}`);
+  }
+
+  const core = HyperlaneCore.fromEnvironment(
+    deployEnvToSdkEnv[environment],
+    multiProvider,
+  );
+
+  for (const chain of core.chains()) {
+    await setStorageGasOracleValues(
+      core,
+      storageGasOracleConfig[chain],
+      chain,
+      args.dryRun,
+    );
+    console.log('\n===========');
+  }
+}
+
+async function setStorageGasOracleValues(
```

Comment:
Out side the scope, but imo this logic would live well in a `HyperlaneApp` for `IGP`?

Reply:
what'd we do for ValidatorAnnounce? Feels like a similar idea, where for now iiuc we've put a lot of that in `infra` but will probably want to rip it out at some point?